### PR TITLE
OPERATOR-262 add cloud_provider support

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:
https://portworx.atlassian.net/browse/OPERATOR-262

**Which issue(s) this PR fixes** (optional)
https://portworx.atlassian.net/browse/OPERATOR-262
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/apis/core/v1/storagecluster.go [hunk 1]
```diff
@@ -387,6 +387,8 @@ type CloudStorageNodeSpec struct {
 
 // CloudStorageCommon details of storage in cloud environment
 type CloudStorageCommon struct {
+	// CloudProviderSpec spec for the cloud provider, such as GKE, AWS etc
+	CloudProviderSpec *string `json:"cloudProviderSpec,omitempty"`
```
Comment (piyush-nimbalkar) [inline pkg/apis/core/v1/storagecluster.go:391]: If this field goes in common spec, it also applies to node spec. I am not sure if we can support multiple cloud providers for a single cluster. Let's confirm that, else we can add this only for cluster level spec.

**Comment by haibinxie** [comment]
@piyush-nimbalkar resolved your comments in a new PR, changed cloud provider to cluster level only.

File: deploy/crds/core_v1_storagecluster_crd.yaml [hunk 2]
```diff
@@ -319,6 +319,9 @@ spec:
               type: object
               description: Details of storage used in cloud environment.
               properties:
+                cloudProvider:
+                  type: string
+                  description: cloud provider name.
```
Comment (piyush-nimbalkar) [inline deploy/crds/core_v1_storagecluster_crd.yaml:324]: nit: Capitalize the doc string - `Cloud provider name`

File: pkg/apis/core/v1/storagecluster.go [hunk 3]
```diff
@@ -360,6 +360,8 @@ type CloudStorageCapacitySpec struct {
 
 // CloudStorageSpec details of storage in cloud environment for entire cluster
 type CloudStorageSpec struct {
+	// CloudProvider spec for the cloud provider, such as GKE, AWS etc
+	CloudProvider *string `json:"cloudProvider,omitempty"`
```
[the remark this answers is not in the document]
Comment (piyush-nimbalkar) [reply pkg/apis/core/v1/storagecluster.go:364]: That makes more sense.

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM after the changes.

**Comment by haibinxie** [comment]
@piyush-nimbalkar @harsh-px resolved all of your comments.

File: pkg/apis/core/v1/storagecluster.go [hunk 4]
```diff
@@ -360,6 +360,8 @@ type CloudStorageCapacitySpec struct {
 
 // CloudStorageSpec details of storage in cloud environment for entire cluster
 type CloudStorageSpec struct {
+	// Provider spec for the cloud provider, such as GKE, AWS etc
+	Provider *string `json:"cloudProvider,omitempty"`
```
Comment (piyush-nimbalkar) [inline pkg/apis/core/v1/storagecluster.go:364]: We need to rename the json field name to `provider`.
Also change the crd schema file above.

Comment (haibinxie) [reply pkg/apis/core/v1/storagecluster.go:364]: Thanks for catching it.

**Review by harsh-px — APPROVED** [review]

**Pull request merged**